front/app/containers/Authentication/SignUpIn/SignUpInModal/SignUp/index.tsx

```diff
@@ -210,11 +210,17 @@ const SignUp = ({
 
   // this automatically completes the 'account-created' step (see stepUtils)
   useEffect(() => {
-    if (activeStep === 'account-created') {
+    if (
+      activeStep === 'account-created' &&
+      !isNilOrError(authUser) &&
+      !authUser.attributes.registration_completed_at
+    ) {
       onCompleteActiveStep();
       setAccountCreated(true);
     }
-  }, [activeStep, onCompleteActiveStep]);
+    // TODO: Temporary fix to prevent multiple calls to onCompleteActiveStep. It's dependencies need to be reviewed as they are causing more calls
```

Comment:
Does it matter that this `useEffect` is called multiple times if you check `!authUser.attributes.registration_completed_at`?

Reply:
Yes it matters. That didn't seem to catch it from my tests. Something is not right with those dependencies. They need more time to investigate. I was pulled into the copy projects work as it is more priority. But also methods will be working on flexible registration so probably not worth investing a lot if time on it.

Reply:
Sure, sounds reasonable.